contracts/storage/RibbonTreasuryVaultStorage.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+abstract contract RibbonTreasuryVaultStorageV1 {
+    // Logic contract used to price options
+    address public optionsPremiumPricer;
+    // Logic contract used to select strike prices
+    address public strikeSelection;
+    // Premium discount on options we are selling (thousandths place: 000 - 999)
+    uint256 public premiumDiscount;
+    // Current oToken premium
+    uint256 public currentOtokenPremium;
+    // Last round id at which the strike was manually overridden
+    uint16 public lastStrikeOverrideRound;
+    // Price last overridden strike set to
+    uint256 public overriddenStrikePrice;
+    // Auction duration
+    uint256 public auctionDuration;
+    // Auction id of current option
+    uint256 public optionAuctionID;
+}
+
+abstract contract RibbonTreasuryVaultStorageV2 {
+    // Amount locked for scheduled withdrawals last week;
+    uint256 public lastQueuedWithdrawAmount;
+}
+
+abstract contract RibbonTreasuryVaultStorageV3 {
+    // Allowed asset for premium denomination
+    mapping(address => bool) allowedAssets;
```

Comment:
Since we're only checking for either WETH or USDC, we can remove the need to store this on storage in the first place

Just check that `premiumAsset == WETH || premiumAsset == USDC`

Reply:
Actually, I don't think we want to limit the type of `premiumAsset` used in the first place

Reply:
So premiumAsset can be any ERC20 tokens?

Reply:
Yep